comms/Attribute.cpp

```diff
@@ -55,6 +55,7 @@ class CAttribute : public CAttributeBase, public IAttribute, public IAttributeHi
     CString m_modifiedBy;
     bool m_eclSet;
     SecAccessFlags m_access;
+    AttrInfo m_attrInfo;
```

Comment:
I don't follow why is this member needed?

Reply:
Before, I had this inside the AttributeToInfo function and it returned the local variable to be used directly and could cause memory problems. So I made it a method variable so that it could remain persistent.

Was:

    AttrInfo AttributeToInfo()
    {
        AttrInfo attrInfo;
        attrInfo.Attribute = this;
        attrInfo.AttributeType = GetType()->GetRepositoryCode();
        return attrInfo;
    }

Now:

    AttrInfo AttributeToInfo()
    {
        m_attrInfo.Attribute = this;
        m_attrInfo.AttributeType = GetType()->GetRepositoryCode();
        return m_attrInfo;
    }

Reply:
Given it was using copy semantics it should have been fine I would have thought?

Reply:
I would have thought so too (and making it a member has problems of its own - in particular not threadsafe, though I don't know if that is an issue in IDE)